compiler-rt/test/asan/TestCases/initialization-nobug-lld.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clangxx_asan -O3 %p/initialization-nobug.cpp %p/Helpers/initialization-nobug-extra.cpp -fuse-ld=lld -o %t && %env_asan_opts=check_initialization_order=true:report_globals=3 %run %t 2>&1 | FileCheck %s --implicit-check-not "DynInit"
```

Comment:
`%S` seems more popular than `%p`

Move this to Linux to ensure the PREINIT_ARRAY available code path.